os/kernel/debug/mem_leak_checker.c

```diff
@@ -246,7 +246,7 @@ static void heap_check(int *leak_cnt)
 					if ((leak_chk >= exclude_bottom && leak_chk <= exclude_top)) {
 						continue;
 					}
-					if (search_hash(*(unsigned long volatile *)leak_chk - (unsigned long)SIZEOF_MM_ALLOCNODE)) {
+					if (search_hash(*(unsigned long volatile *)leak_chk)) {
```

Comment:
I think this change is wrong. This logic is very similar to checking the data sections (like .data and .bss)  https://github.com/Samsung/TizenRT/blob/a67b9f7f062b443d4f778117f720af50ea557263/os/kernel/debug/mem_leak_checker.c#L215

Since, here, we are checking the contents of the node, we need to subtract SIZEOF_MM_ALLOCNODE to get the pointer to the alloc_node.

we are checking for cases like this,
```
int ** a;
a = malloc((sizeof(int *)) * 10);
for (int i = 0; i < 10; i++) a[i] = malloc(sizeof(int));
```

alloc node of a contains the pointers of a[0], a[1]..... , 
but the value of alloc node of a[0] which is present in the hash table  = a[0] - SIZEOF_MM_ALLOCNODE

Reply:
yes, right.
I too found from my tests that this is the correct code
Thank you for reiew~

I am facing problem that detects wrong mem leak.
some memories that were previously confirmed as memory in the data and BSS areas are being detected as memory leaks.
As you reviewed, this code would not be the cause of the problem.
let me check more~ , Thank you~